Profile page added

Damir-Mu opened the pull request.

Damir-Mu committed:
Profile page added

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+
+    useEffect(() => {
+
+        if (currentUser?.photoURL) {
```
And what if user decides to leave avatar blanked? You can't rely on this, as we are not forcing selection of the photo at the registration

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+        if (currentUser?.photoURL) {
+            setAvatarUrl(currentUser.photoURL);
+        }
+        setUserName(currentUser.displayName);
```
No user = no displayName = app crush. Change the way you are applying the useEffect, it shouldn't dependent on the fields but on the object itself. You are setting photoUrl as a dependency but again it will only be triggered if there was a photo before.

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+        }
+        setUserName(currentUser.displayName);
+
+    }, [currentUser.photoURL, currentUser.displayName])
```
Search the docs to return unsubcriber and clean the hook trigger

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+
+    const [changeData, setChangeData] = useState({
+        nickname: ""
+        //  changed data object
```
No need to double the state for the nickname. Setting one as an initial for the existing name is enough, updating the state on it won't yet trigger an update at the db.

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+    const handleNameChange = (event) => {
+        const { name, value } = event.target;
+
+        setChangeData({
```
setUserName(event.target.value)

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+
+    const handleNameChangeSubmit = (event) => {
+        event.preventDefault();
+        updateProfile(currentUser, { displayName: changeData.nickname });
```
await updateProfile(currentUser, { displayName: userName });

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+        updateProfile(currentUser, { displayName: changeData.nickname });
+        if (event.target[0].value) {
+            event.target[0].value = '';
+        }
```
That's meaningless. The form should represent the latest value of the displayName

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+        return '';
+    };
+
+
```
Already commented before. User shouldn't have an ability to change the only contact method in existence.

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+
+        setErrorEmailAuth(newError);
+
+        if (newError.length === 0) {
```
Any symbol inside string will return true. if(newError)

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+
+        if (newError.length === 0) {
+            // call submit function
+            sendPasswordResetEmail(auth, emailAuth)
```
Why send resetEmail? If user reached this page he never forgot password, just want to change it

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+                    >
+                        <Tab eventKey="desc" title="Profile">
+                            <div className="profileAvatar">
+                                <label id='avatarInput'>
```
Inconsistent format. Every tab has a form to change data but this one.

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+    // default avatar png       
+    const [avatarUrl, setAvatarUrl] = useState(AddImg);
+    const [photoBinary, setPhotoBinary] = useState(null);
+    const [loading, setLoading] = useState(false);
```
loading never used

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+                            </div>
+                            <div className="profileDesc">
+                                <span>
+                                    {/*  Add description part to document write functionality  */}
```
So add the description, placeholder confuses

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+                            <div className="profileAvatar">
+                                <label id='avatarInput'>
+                                    <img src={avatarUrl} alt="avatar" id='avatarImage' />
+                                    <input type="file" hidden={true} onChange={handleChange} />
```
Label without description. It is unclear what this photo serves for

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -0,0 +1,216 @@
+        // accessing the url of the photo
+        const photoURL = await getDownloadURL(fileRef)
+
+        updateProfile(currentUser, { photoURL });
```
Async operation

DenisTagaev commented on src/pages/profile/Profile.scss:
```diff
@@ -0,0 +1,71 @@
+            color: rgb(187, 1, 1);
+        }
+    }
+}
```
Overall please comment the function's purpose. Comments like 'call submit are not descriptive'. Regroup the code so that useStates are together, functions in the order of use, pay attention to async operations. No hints regarding what's going on on the screen like at least what forms are for? What is the current data? Functions are only updating the 'user' collection, what if there are 'userChats' for the user? Validation to be corrected.

Damir-Mu committed:
Profile update 1.1

Still huge amount of work left

Damir-Mu committed:
Redone UseEffect, joined all errors

Besides adding the password change form, which needs to be finished. However, can not resolve an error at the moment. Error in try catch at line 189

Damir-Mu committed:
Update Profile.jsx

Damir-Mu committed:
Merge branch 'main' of github.com:DenisTagaev/webchat into damir-Mu

Damir-Mu committed:
Profile page updated, styles added

Description for profile left.
But basic skeleton is done

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -19,141 +23,234 @@ import Tabs from 'react-bootstrap/Tabs';
+        const unsub = setUserName(currentUser.displayName);
+        return () => {
+            console.log("re-rendered")
+            return unsub;
```
unsub()

DenisTagaev commented on src/pages/profile/Profile.jsx:
```diff
@@ -19,141 +23,234 @@ import Tabs from 'react-bootstrap/Tabs';
 
-        setErrorEmailAuth(validateEmailField(value));
+    const handleNewPasswordSubmit = () => {
+        console.log(newPassword)
```
Async operation

Damir-Mu committed:
Fixed profile

Bootstrap cut

Damir-Mu committed:
Tabs - replaced

Damir-Mu committed:
Bootstrap package removed

Damir-Mu committed:
Slight, cleaning

Damir-Mu committed:
Asyncs cleaned

Quite sleepy, therefore so many commits

Damir-Mu committed:
Update Profile.jsx

Damir-Mu committed:
Profile useefect update

Damir-Mu committed:
Description profile update

Damir-Mu committed:
Merge branch 'main' of github.com:DenisTagaev/webchat into damir-Mu

Damir-Mu committed:
Added avatar prompt component

Rework in progress

Damir-Mu committed:
profile

Working with description display

Damir-Mu committed:
Not found in progress

Damir-Mu committed:
Description detailed

More to add

Damir-Mu committed:
package-lock change

Damir-Mu committed:
Added bugreport form. changed styling and display.

Fixed previous problems
